pkg/adapter/serializer/effectiveBlueprintV1/effectiveBlueprintV1.go

```diff
@@ -0,0 +1,147 @@
+	errorList = append(errorList, doguError, componentError)
+
+	err := errors.Join(errorList...)
+	if err != nil {
+		return EffectiveBlueprintV1{}, fmt.Errorf("cannot convert blueprintMask to BlueprintMaskV1 DTO: %w", err)
+	}
+
+	registryConfigAbsent := blueprint.RegistryConfigAbsent
+	if registryConfigAbsent == nil {
+		registryConfigAbsent = []string{}
+	}
+
+	return EffectiveBlueprintV1{
+		Dogus:                   convertedDogus,
+		Components:              convertedComponents,
+		RegistryConfig:          flattenRegistryConfig(blueprint.RegistryConfig),
+		RegistryConfigAbsent:    registryConfigAbsent,
+		RegistryConfigEncrypted: flattenRegistryConfig(blueprint.RegistryConfigEncrypted),
+	}, nil
+}
+
+func ConvertToEffectiveBlueprint(blueprint EffectiveBlueprintV1) (domain.EffectiveBlueprint, error) {
+	convertedDogus, doguErr := serializer.ConvertDogus(blueprint.Dogus)
+	convertedComponents, compErr := serializer.ConvertComponents(blueprint.Components)
+	convertedConfig, configError := convertToRegistryConfig(blueprint.RegistryConfig)
+	convertedEncryptedConfig, encryptedConfigError := convertToRegistryConfig(blueprint.RegistryConfig)
+	err := errors.Join(doguErr, compErr, configError, encryptedConfigError)
+	if err != nil {
+		return domain.EffectiveBlueprint{}, fmt.Errorf("syntax of blueprintV2 is not correct: %w", err)
+	}
+	return domain.EffectiveBlueprint{
+		Dogus:                   convertedDogus,
+		Components:              convertedComponents,
+		RegistryConfig:          convertedConfig,
+		RegistryConfigAbsent:    blueprint.RegistryConfigAbsent,
+		RegistryConfigEncrypted: convertedEncryptedConfig,
+	}, nil
+}
+
+func convertToRegistryConfig(flattenedConfig map[string]string) (domain.RegistryConfig, error) {
+	// expand key structure
+	widenedMap := widenMap(flattenedConfig)
+	//convert it to domain.RegistryConfig (which has at least depth 2)
+	config := domain.RegistryConfig{}
+	for key1, val1 := range widenedMap {
+		switch subMap := val1.(type) {
+		case map[string]interface{}:
+			for key2, val2 := range subMap {
+				if config[key1] == nil {
+					config[key1] = make(map[string]interface{})
+				}
+				config[key1][key2] = val2
+			}
+		default:
+			return domain.RegistryConfig{}, fmt.Errorf("registry config is invalid: values need to be at least at depth 2: key %v is invalid", key1)
+		}
+	}
+	return config, nil
+}
+
+func widenMap(currentMap map[string]string) map[string]interface{} {
+	newMap := map[string]interface{}{}
+	for key, val := range currentMap {
+		keys := strings.Split(key, configKeySeparator)
+		setKey(keys, val, newMap)
+	}
+	return newMap
+}
+
+func setKey(keys []string, value string, initialMap map[string]interface{}) {
+	currentMap := initialMap
+	length := len(keys)
+	for i, key := range keys {
+		if i == length-1 {
+			currentMap[key] = value
+			break
+		}
+		if currentMap[key] == nil {
+			currentMap[key] = map[string]interface{}{}
+		}
+		currentMap = currentMap[key].(map[string]interface{})
```

Comment:
🚫 **[checkstyle]** <forcetypeassert> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>type assertion must be checked